src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
+    class Node<T> {
+        private T element;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T element) {
+            this.element = element;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (first == null) {
+            addToEmptyList(value);
+        } else {
+            addLastNode(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        if (first == null) {
+            addToEmptyList(value);
+        }
+        if (index == 0) {
+            addToFirstIndex(value);
+        } else {
+            addToMiddleIndex(value, index);
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNodeByIndex(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNodeByIndex(index);
+        T oldValue = node.element;
+        node.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> removedElement = getNodeByIndex(index);
+        unLink(removedElement);
+        size--;
+        return (T) removedElement;
     }
 
     @Override
     public boolean remove(T object) {
+        int index = -1;
+        for (int i = 0; i < size; i++) {
+            if ((object != null && get(i) != null && get(i).equals(object))
+                    || (object == null && get(i) == null)) {
+                index = i;
+                break;
+            }
+        }
+        if (index != -1) {
+            Node<T> removedNode = getNodeByIndex(index);
+            unLink(removedNode);
+            return true;
+        }
         return false;
```

Comment:
you dont'need that technique with the variable here (flag if found). Just call unlink on line 85 and return true, otherwise return false outside of the loop